.github/workflows/ui-tests-common.yml

```diff
@@ -0,0 +1,103 @@
+name: UI Tests Common
+
+on:
+  workflow_call:
+    inputs:
+      fastlane_command:
+        description: "The fastlane command to run"
+        required: true
+        type: string
+      xcresult_name:
```

Comment:
We could probably get away without using this, and just form a name based on the other inputs to this workflow: `uitests-xcresult-$fastlane_command-$xcode_version-$macos_version`, wdyt?

Reply:
Looking again I see that one of the values has spaces, so we'd have to write a slugifier :/

Reply:
Can also just use some internal formulation, since the xcresult is only really needed for the `Publish Test Report` step in this reusable workflow.

Reply:
I feel this can be reused, but there seem to be some different names, sometimes dashes are replaced by underscore.

Reply:
I think I got it in [01739d0](https://github.com/getsentry/sentry-cocoa/pull/5375/commits/01739d094809ee042556730c93991b7e8fb6c5da), I thought the name was automatic